### PR TITLE
Ag/rm new subscription btn

### LINKED ISSUE #28987: Remove `New subscription button` from subscriptions page
As part of [this proposal](https://docs.google.com/document/d/1XBIJkarIk6b9Kve311X6N6TCMAdyFM0DOi_iWbmCcQc/edit#), we'd like to Remove the `New subscription` button from the subscriptions page (Update Settings > Subscriptions) and update the copy to read, “Contact us to purchase a subscription for a self-hosted Sourcegraph instance. See [pricing](https://about.sourcegraph.com/pricing/) for more information.” Reference below:

28987 - Subscription page updates

### PR DESCRIPTION
**Pull request by anthonygedeon**
## Context
Removes the `New Subscription` button and updates the body copy of the product component

## Result
image

Closes #28987

### CODE REVIEW

**Comment by anthonygedeon** [comment]
@sourcegraph/contribution-reviewers

File: client/web/src/enterprise/user/productSubscriptions/UserSubscriptionsProductSubscriptionsPage.tsx [hunk 1]
```diff
@@ -80,14 +80,9 @@ export const UserSubscriptionsProductSubscriptionsPage: React.FunctionComponent<
...
                 path={[{ text: 'Subscriptions' }]}
-                actions={
-                    <Link to={`${props.match.path}/new`} className="btn btn-primary text-nowrap">
-                        New subscription
-                    </Link>
-                }
                 description={
                     <>
-                        Purchase a subscription for a self-hosted Sourcegraph instance. See{' '}
+                        Conatct us to purchase a subscription for a self-hosted Sourcegraph instance. See{' '}
```
Comment (oleggromov) [inline client/web/src/enterprise/user/productSubscriptions/UserSubscriptionsProductSubscriptionsPage.tsx:85]: There's a typo:
```suggestion
                        Contact us to purchase a subscription for a self-hosted Sourcegraph instance. See{' '}
```

File: yarn.lock [hunk 2]
```diff
@@ -6430,7 +6430,7 @@ any-promise@^1.0.0, any-promise@^1.1.0, any-promise@~1.3.0:
   resolved "https://registry.npmjs.org/any-promise/-/any-promise-1.3.0.tgz#abc6afeedcea52e809cdc0376aed3ce39635d17f"
   integrity sha1-q8av7tzqUugJzcA3au0845Y10X8=
 
-anymatch@*, anymatch@^3.0.3, anymatch@~3.1.1, anymatch@~3.1.2:
+anymatch@*, anymatch@^3.0.3, anymatch@~3.1.1:
```
Comment (oleggromov) [inline yarn.lock:6433]: I think changes to `yarn.lock` shouldn't be part of this change as we a) try to keep PRs related to a single change and b) aim to update dependencies in an organized way.

Please let me know if updating to the latest main and running `yarn` are causing these - and in any case, please remove changes from this file from the PR.

**Review by oleggromov — CHANGES_REQUESTED** [review]
There's a copy change (typo) I requested.
Other than that, and suggested removal of changes to the `yarn.lock` file, it's all good. 

The reason why you aren't able to see this page/button locally is because it's required to be logged in. Not sure if that's possible via `sg start oss` - will check. Regardless, the change is correct. Thanks for your contribution! 👍

**Comment by oleggromov** [comment]
Due to inactivity here, I am closing this PR and opening a new one without the mentioned issues: https://github.com/sourcegraph/sourcegraph/pull/30490

**Pull request closed without merging**